src/lua/msgpack.c

```diff
@@ -809,6 +811,187 @@ luamp_object_iterator(struct lua_State *L)
 	return 1;
 }
 
+/**
+ * Helper for indexing `msgpack.object` in the case, when the indexation key
+ * is an integer. The underlying MsgPack is expected to be an array.
+ */
+static int
+luamp_object_get_integer_array(struct lua_State *L, struct luamp_object *obj,
+			       uint64_t idx, bool is_signed)
+{
+	if (is_signed)
+		return 0;
+
+	const char *it = obj->data;
+	uint32_t size = mp_decode_array(&it);
+	if (idx == 0 || idx > size) {
+		return 0;
+	}
+	for (; idx > 1; --idx)
+		mp_next(&it);
+	luamp_decode(L, obj->cfg, &it);
+	return 1;
+}
+
+/**
+ * Helper for indexing `msgpack.object` in the case, when the indexation key
+ * is an integer. The underlying MsgPack is expected to be a map.
+ */
+static int
+luamp_object_get_integer_map(struct lua_State *L, struct luamp_object *obj,
+			     uint64_t key, bool is_signed)
+{
+	const char *it = obj->data;
+	uint32_t size = mp_decode_map(&it);
+	for (uint32_t i = 0; i < size; ++i, mp_next(&it)) {
+		switch (mp_typeof(*it)) {
+		case MP_UINT: {
+			uint64_t probe = mp_decode_uint(&it);
+			if (!is_signed && key == probe) {
+				luamp_decode(L, obj->cfg, &it);
+				return 1;
+			}
+			break;
+		}
+		case MP_INT: {
+			int64_t probe = mp_decode_int(&it);
+			if (is_signed) {
+				if ((int64_t)key == probe) {
+					luamp_decode(L, obj->cfg, &it);
+					return 1;
+				}
+			} else {
+				if (probe >= 0 && key == (uint64_t)probe) {
```

Comment:
This code isn't covered: https://coveralls.io/jobs/110054477/source_files/28263573271#L864

Reply:
Fixed: added a test-case for this branch.